src/util/bip32.rs

```diff
@@ -161,6 +161,21 @@ impl fmt::Display for ChildNumber {
     }
 }
 
+impl FromStr for ChildNumber {
+    type Err = Error;
+
+    fn from_str(inp: &str) -> Result<ChildNumber, Error> {
+        Ok(match inp.chars().last().map_or(false, |l| l == '\'' || l == 'h') {
+            true => ChildNumber::from_hardened_idx(
+                inp[0..inp.len() - 1].parse().map_err(|_| Error::InvalidChildNumberFormat)?
+            ),
+            false => ChildNumber::from_normal_idx(
```

Comment:
I think this could panic with `"ChildNumber indices have to be within [0, 2^31 - 1], is: {}"` (in `bitcoin::util::bip32::ChildNumber::from_normal_idx`).

Test case: `DerivationPath::from_str("m/2147483648");`

This should be an error instead since `FromStr` could be used with untrusted input.